crossing-repair.lic

```diff
@@ -33,6 +87,80 @@ class CrossingRepair
+    # repair is deprecated, we still need to support the legacy settings.
+    leather_items_to_repair = @equipment_manager.items.select(&:leather).reject(&:skip_repair)
+    metal_items_to_repair = @equipment_manager.items.reject(&:leather).reject(&:skip_repair)
+
+    # No bank specified in base-town.yaml for specified repair town? Message user. This method
+    # both withdraws and deposits coins, so having a bank specified is requisite.
+
+    unless @skip_bank && DRCM.wealth(settings['hometown'])
+      unless @hometown_data['deposit']
+        DRC.message("No bank found for #{town} in base-town.yaml. Please submit an issue on GitHub:")
+        DRC.message("https://github.com/rpherbig/dr-scripts/issues")
+        exit
+      end
+    end
+
+    # Check if we have leather items specified in yaml. If so, verify we have a leather
+    # repair person specified in base-town.yaml for the user-specified repair town.
+    # If not, message user and direct to report issue.
+    if leather_items_to_repair.any?
+      echo "Leather items detected." if $debug_mode_crossing_repair
+      unless @hometown_data['leather_repair']
+        DRC.message("No leather repairer specified in base-town.yaml. Please submita an issue on GitHub:")
+        DRC.message("https://github.com/rpherbig/dr-scripts/issues")
+        exit
+      end
+    end
+
+    # Check if we have metal items specified in yaml. If so, verify we have a metal
+    # repair person specified in base-town.yaml for the user-specified repair town.
+    # If not, message user and direct to report issue.
+    if metal_items_to_repair.any?
+      echo "Metal items detected." if $debug_mode_crossing_repair
+      unless @hometown_data['metal_repair']
+        DRC.message("No metal repairer specified in base-town.yaml. Please submita an issue on GitHub:")
+        DRC.message("https://github.com/rpherbig/dr-scripts/issues")
+        exit
+      end
+    end
+
+    # Verify we have enough money as specified in the user's yaml
+    # Also start the performance script for some lovely extra exp = tdps
+    DRCM.ensure_copper_on_hand(@repair_withdrawal_amount, settings)
+    start_script('performance', ['noclean']) unless settings.instrument
+
+    # Display our metal and leather item hashes for repair if debug set
+    if $debug_mode_crossing_repair
+      DRC.message("Metal items hash:")
+      echo @hometown_data['metal_repair'] => metal_items_to_repair
+
+      DRC.message("Leather items hash:")
+      echo @hometown_data['leather_repair'] => leather_items_to_repair
+    end
+
+    # If we've not exited above, we're validated ready to repair.
+    # Populate a hash of items to repair at either leather or metal repairer.
+    # Return this hash for future use.
+    repair_info = { @hometown_data['leather_repair'] => leather_items_to_repair }
+                    .merge(@hometown_data['metal_repair'] => metal_items_to_repair) { |_key, old, new| old + new }
```

Comment:
🧹 [Rubocop](https://docs.rubocop.org/rubocop/1.25/index.html), a static code analyzer and linter for Ruby, prefixes unused arguments with an underscore `_` as some convention to say it's unused. We don't use Rubocop in this project and I think the underscores add to confusion for new developers. While you're in here, do you mind removing this underscore? Thanks
```suggestion
                    .merge(@hometown_data['metal_repair'] => metal_items_to_repair) { |key, old, new| old + new }
```

Reply:
Honestly, I prefer the underscores to make it clear that something isn't being used.

And we did use Rubocop for many years, but it had to be run manually and it's been quite a while since I ran it. If someone wanted to run it (with autofix enabled) it would be appreciated. It lends a nice consistency to the codebase.

Reply:
I don't understand how `key` is unused in this scenario.

Reply:
Ok, I guess I just misunderstand the purpose. I can't leave off the `key` argument, right? So I guess Rubocop still marks it if it's "unused" but required?

Reply:
Correct, you can't omit the argument since it's still provided to the block of code by Ruby

Reply:
Settled, per comments above.